File: AgentFramework/dotnet/AzureAICommunityAgent/Middleware/YouTubeVideoMiddleware/YouTubeConfig.cs
using Microsoft.Extensions.Logging;

namespace AzureAICommunity.Agent.Middleware.YouTube;

/// <summary>
/// Holds all configuration settings required to connect to and query the YouTube Data API.
/// Pass an instance of this class to <see cref="YouTubeTools"/> or <see cref="YouTubeFetch"/>
/// to control API credentials, target channel, and result-count limits.
/// </summary>
public sealed class YouTubeConfig
{
    /// <summary>YouTube Data API v3 key used to authenticate requests.</summary>
    public string ApiKey { get; init; } = string.Empty;

    /// <summary>Optional YouTube channel ID to restrict search results to a specific channel.</summary>
    public string ChannelId { get; init; } = string.Empty;

    /// <summary>Upper bound on the number of results the API may return per request.</summary>
    public int MaxResults { get; init; } = 25;

    /// <summary>Default number of videos to return when the caller does not specify a count.</summary>
    public int DefaultCount { get; init; } = 10;

    /// <summary>Optional logger factory used to create loggers throughout the middleware pipeline.</summary>
    public ILoggerFactory? LoggerFactory { get; init; }
}

File: AgentFramework/Python/Module/YouTubeModule/youtube_search_module/youtube_config.py
from __future__ import annotations

import logging
from dataclasses import dataclass


@dataclass
class YouTubeConfig:
    """Holds all configuration settings required to connect to and query the YouTube Data API v3."""

    api_key: str
    """YouTube Data API v3 key used to authenticate requests."""

    channel_id: str = ""
    """Optional YouTube channel ID to restrict search results to a specific channel."""

    max_results: int = 25
    """Upper bound on the number of results the API may return per request."""

    default_count: int = 10
    """Default number of videos to return when the caller does not specify a count."""

    logger: logging.Logger | None = None
    """Optional logger instance. Falls back to logging.getLogger(__name__) when not provided."""
